gobblin-hive-registration/src/main/java/gobblin/hive/policy/HiveRegistrationPolicyBase.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.hive.policy;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+
+import gobblin.annotation.Alpha;
+import gobblin.configuration.State;
+import gobblin.hive.HivePartition;
+import gobblin.hive.spec.HiveSpec;
+import gobblin.hive.spec.SimpleHiveSpec;
+
+
+/**
+ * A base implementation of {@link HiveRegistrationPolicy}. It obtains database name from
+ * property {@link #HIVE_DATABASE_NAME} or {@link #HIVE_DATABASE_REGEX} (group 1), obtains
+ * table name from property {@link #HIVE_TABLE_NAME} and {@link #HIVE_TABLE_REGEX} (group 1),
+ * and builds a {@link SimpleHiveSpec}.
+ *
+ * @author ziliu
+ */
+@Alpha
+public abstract class HiveRegistrationPolicyBase implements HiveRegistrationPolicy {
```

Comment:
I would suggest splitting this into two classes one for `HiveRegisrationPolicyFromState` and `HiveRegistrationPolicyFromRegex`. The former checks the given `State` to see if the db and table are specified, the later uses a regex to find the db and table.

Is there a concrete use case for having a policy where the state is first checked, and then a regex is used?